example/presto_cluster/main.tf

```
@@ -31,43 +48,54 @@ module "presto" {
   use_canary       = true
 
   # other
-  minio = {
+  hivemetastore_service = {
+    service_name = module.hive.service_name
+    port         = module.hive.port
+  }
+
+  minio_service = {
     service_name = module.minio.minio_service_name
     port         = module.minio.minio_port
     access_key   = module.minio.minio_access_key
     secret_key   = module.minio.minio_secret_key
   }
-  hivemetastore = {
-    service_name = module.hive.service_name
-    port         = module.hive.port
-  }
+
+  # Vault provided credentials
+  minio_vault_secret = local.minio_vault_secret
+
 }
 
 module "minio" {
   source = "github.com/fredrikhgrelland/terraform-nomad-minio.git?ref=0.3.0"
 
   # nomad
-  nomad_datacenters = local.nomad_datacenters
-  nomad_namespace   = local.nomad_namespace
+  nomad_datacenters = ["dc1"]
+  nomad_namespace   = "default"
   nomad_host_volume = "persistence-minio"
 
   # minio
   service_name    = "minio"
   host            = "127.0.0.1"
   port            = 9000
   container_image = "minio/minio:latest" # todo: avoid using tag latest in future releases
+
+  # Vault provided credentials
+  # todo: follow naming convention issue https://github.com/fredrikhgrelland/terraform-nomad-minio/issues/87
   vault_secret = {
-    use_vault_provider   = false,
-    vault_kv_policy_name = "",
-    vault_kv_path        = "",
-    vault_kv_access_key  = "",
-    vault_kv_secret_key  = ""
+    use_vault_provider   = local.minio_vault_secret.use_vault_provider
+    vault_kv_policy_name = local.minio_vault_secret.vault_kv_policy_name
+    vault_kv_path        = local.minio_vault_secret.vault_kv_path
+    vault_kv_access_key  = local.minio_vault_secret.vault_kv_access_key_name
+    vault_kv_secret_key  = local.minio_vault_secret.vault_kv_secret_key_name
```

Is there any reason you added this to local instead of writing it explicitly?
```suggestion
    use_vault_provider   = true
    vault_kv_policy_name = "kv-secret"
    vault_kv_path        = "secret/data/dev/minio"
    vault_kv_access_key  = "access_key"
    vault_kv_secret_key  = "secret_key"
```

---

Ref the question above